test/WorkerExtensionTests/Blob/BlobStorageConverterTests.cs

```diff
@@ -22,375 +24,773 @@
+            Assert.IsType<String[]>(stringResult);
+            Assert.Equal("MyBlobString", stringResult.First().ToString());
         }
 
         [Fact]
-        public async Task ConvertFromBindingDataAsync_ReturnsUnhandled()
+        public async Task ConvertAsync_ValidModelBindingData_Stream_ReturnsSuccess()
         {
-            object source = GetTestGrpcModelBindingData(GetFullTestBinaryData());
-            var context = new TestConverterContext(typeof(string), source);
-            var dict = _mockBlobStorageConverter.Object.GetBindingDataContent((Worker.Core.ModelBindingData)source);
+            // Arrange
+            var grpcModelBindingData = Helper.GetTestGrpcModelBindingData(GetTestBinaryData(), "AzureStorageBlobs");
+            var context = new TestConverterContext(typeof(Stream), grpcModelBindingData);
 
-            _mockBlobStorageConverter.Setup(c => c.ConvertModelBindingDataAsync(dict, typeof(string), (Worker.Core.ModelBindingData)source)).ReturnsAsync(null);
+            var expectedStream = new MemoryStream(Encoding.UTF8.GetBytes("MyBlobStream"));
+            var blobDownloadResult = BlobsModelFactory.BlobDownloadStreamingResult(expectedStream);
+            var mockResponse = new Mock<Response<BlobDownloadStreamingResult>>();
+            mockResponse.SetupGet(r => r.Value).Returns(blobDownloadResult);
 
-            var conversionResult = await _mockBlobStorageConverter.Object.ConvertFromBindingDataAsync(context, (Worker.Core.ModelBindingData)source);
+            var mockBlobClient = new Mock<BlobClient>();
+            mockBlobClient
+                .Setup(m => m.DownloadStreamingAsync(It.IsAny<HttpRange>(), It.IsAny<BlobRequestConditions>(), It.IsAny<bool>(), null, default))
+                .ReturnsAsync(mockResponse.Object);
 
-            Assert.Equal(ConversionStatus.Unhandled, conversionResult.Status);
+            var mockContainer = new Mock<BlobContainerClient>();
+            mockContainer.Setup(m => m.GetBlobClient(It.IsAny<string>())).Returns(mockBlobClient.Object);
+
+            _mockBlobServiceClient.Setup(m => m.GetBlobContainerClient(It.IsAny<string>())).Returns(mockContainer.Object);
+
+            // Act
+            var conversionResult = await _blobStorageConverter.ConvertAsync(context);
+            var streamResult = (Stream)conversionResult.Value;
+
+            // Assert
+            Assert.Equal(ConversionStatus.Succeeded, conversionResult.Status);
+            Assert.Equal(expectedStream, streamResult);
         }
 
-        [Theory]
-        [InlineData(Constants.BlobExtensionName, true)]
-        [InlineData(" ", false)]
-        [InlineData("incorrect-value", false)]
-        public void IsBlobExtension_MatchesExpectedOutput(string sourceVal, bool expectedResult)
+        [Fact (Skip = "Fails - broken scenario. ChangeType fails trying to convert MemoryStream to Stream) - 'Object must implement IConvertible'")]
```

Comment:
@fabiocav same situation as above but the difference here is that **this is a real broken scenario** (the others only fail because we are mocking them). Also hoping your changes will fix this with the removal of ChangeType